numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -4166,6 +4166,55 @@ small_correlate(const char * d_, npy_intp dstride,
     }
 }
 
+/*
+*/
+
+/* A clone function for the datetime dtype metadata */
+static NpyAuxData *
+_datetime_dtype_metadata_clone(NpyAuxData *data)
+{
+    PyArray_DatetimeDTypeMetaData *newdata =
+        (PyArray_DatetimeDTypeMetaData *)PyArray_malloc(
+                        sizeof(PyArray_DatetimeDTypeMetaData));
+    if (newdata == NULL) {
+        return NULL;
```

Comment:
`return PyErr_NoMemory()` for good measure here too.

Reply:
added

Reply:
But its signature is to return `PyObject*`. Couldn't chaining the return cause a warning?

Reply:
You're correct, good point